core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -127,7 +134,7 @@ private[spark] abstract class MapOutputTracker(conf: SparkConf) extends Logging
    * Called from executors to get the server URIs and output sizes of the map outputs of
    * a given shuffle.
    */
-  def getServerStatuses(shuffleId: Int, reduceId: Int): Array[(BlockManagerId, Long)] = {
+  def getServerStatuses(shuffleId: Int, reduceId: Int): Array[ServerAttemptSize] = {
```

Comment:
this change is just adding `stageAttemptId` to the return type.  Then I wrapped in a case class instead of a tuple b/c it was easy to get confused what each member was.